### PR TITLE
Composite basis

### PR DESCRIPTION
**Pull request by shinaoka**
* Implemented composite basis
* Introduced several basis sets for augmentation (LegendreBasis, MatsubaraConstBasis)

### CODE REVIEW

File: src/irbasis3/augmentation.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            data[l, 0, l] = np.sqrt((l+0.5)/beta) * cl[l]
+        self.u = PiecewiseLegendrePoly(data, knots)
+
+        # self.uhat
+        # Hack: See basis.py
+        uhat_base = PiecewiseLegendrePoly(np.sqrt(beta) * self.u.data, np.array([-1,1]))
+        self.uhat = uhat_base.hat({'F': 'odd', 'B': 'even'}[statistics])
+
+        # self.v
+        self.v = None
```
Comment (mwallerb) [inline src/irbasis3/augmentation.py:47]: The Legendre basis is the limit $Lambda -> 0$ of the IR basis.

So, in principle we could add `s` and `v` here (`s[0]` is just 1/2, `s[1:]` is zero, `v` is again simply a Legendre basis) ... the disadvantage is that while the interface is then consistent, the values are pretty much useless for analytic continuation.

What do you think?

File: src/irbasis3/sampling.py [hunk 2]
```diff
@@ -23,16 +23,16 @@ class SamplingBase:
...
-            sampling_points = self.__class__.default_sampling_points(basis)
-        else:
-            sampling_points = np.array(sampling_points)
-
+    def __init__(self, basis, sampling_points):
+        #sampling_points = np.array(sampling_points)
+        #if sampling_points is None:
+            #sampling_points = self.__class__.default_sampling_points(basis)
+        #else:
+            #sampling_points = np.array(sampling_points)
```
Comment (mwallerb) [inline src/irbasis3/sampling.py]: Suggest removing commented-out code
```suggestion
```

File: src/irbasis3/augmentation.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            # self.uhat
+            self.uhat = _ConstTerm(value)
+    
+            # self.v
+            self.v = None
+    
+            # Default sampling points
+            self.default_tau_sampling_points = np.array([])
+            self.default_matsubara_sampling_points = np.array([])
+class _ConstTerm:
```
Comment (mwallerb) [inline src/irbasis3/augmentation.py:87]:
```suggestion


class _ConstTerm:
```

**Review by mwallerb — APPROVED** [review]
Looks good!

File: src/irbasis3/augmentation.py (hunk 1 above)
Comment (shinaoka) [reply src/irbasis3/augmentation.py:47]: Either is fine for me. Shall we keep it as it is and improve it after we merge the PR?

Comment (mwallerb) [reply src/irbasis3/augmentation.py:47]: Works for me! :)

**Comment by shinaoka** [comment]
We have agreed to merge this PR (and improve the code afterward).

**Pull request merged**